File: Round-1/Array/Running Sum of 1d Array/runningSum.cpp
#include <iostream>
#include <vector>
using namespace std;
class Solution
{
public:
    vector<int> runningSum(vector<int> &nums)
    {
        unsigned long int count = 0;
        for (int i = 0; i < nums.size(); ++i)
        {
            count += nums[i];
            nums[i] = count;
        }
        return nums;
    }
};
int main()
{
    Solution sol;
    vector<int> arr{1, 2, 3, 4, 5, 6};
    sol.runningSum(arr);
    return 0;
}

File: Round-1/Array/Running Sum of 1d Array/runningSum.js
//Sol - 1 >> Start of brute force approach to solve problem - wrost time complexity O(n^2)
const runningSum = (nums) => {
    let count = 0;
    let resultArray = [];
    for (let j = 0; j < nums.length; j++) {
        count = 0;
        for (let i = 0; i <= j; i++) {
            count += nums[i]
        }
        resultArray.push(count);
    }
    return resultArray;
}

console.log(runningSum([1, 22, 4, 5, 6, 7, 4, 6, 9]));
// End of brute force


//Sol - 2 >> Start some optimal solution - wrost time complexity O(n);
const runningSum = (nums) => {
    let sum = 0;
    for (i = 0; i < nums.length; i++) {
        sum += nums[i];
        nums[i] = sum;
    }
    return nums;
}

console.log(runningSum([1, 22, 4, 5, 6, 7, 4, 6, 9]));
// End